src/view/ImageView.cxx

```diff
@@ -0,0 +1,58 @@
+#include "ImageView.hxx"
+#include "../renderer/iRenderer.hxx"
+#include "../util/PixelBuffer.hxx"
+#include "../util/Filesystem.hxx"
+#include "../util/LOG.hxx"
+#include "../util/Exception.hxx"
+#include <vector>
+
+ImageView::ImageView(std::string && image, DisplayMode mode) : 
+  m_ImagePath(fs::getBasePath() + image + ".png"),
+  m_DisplayMode(mode)
+{ }
+
+ImageView::~ImageView() { }
+
+void ImageView::draw(iRenderer & renderer) const noexcept
+{
+  Rectangle target = getBounds();
+  PixelBuffer pb = fs::readPNGFile(m_ImagePath);
+  switch(m_DisplayMode)
+  {
+    case DisplayMode::NearestXY:
+      nearestXY(pb, target);
+      break;
+    case DisplayMode::RepeatXY:
+      repeatXY(pb, target);
+  }
+  if(pb.isEmpty())
+  {
+    return;
+  }
+  renderer.drawPicture(target, pb.data());
+}
+
+void ImageView::nearestXY(PixelBuffer & pb, Rectangle & target) const noexcept
+{
+  // Preserve aspect ratio
+  Rectangle o_bounds = pb.bounds();
+  float factor = std::min(o_bounds.width() / target.width(), 
+        o_bounds.height() / target.height());
+  Rectangle nearest = pb.bounds();
+  // Center nearest within target
+  nearest.translateX((target.width() - nearest.width()) / 2 + target.x1());
+  nearest.translateY((target.height() - nearest.height()) / 2 + target.y1());
```

Comment:
I do not believe it really centres nearest. I suppose **- nearest.x1()** is missing.

Reply:
nearest.x1() == 0